crates/biome_js_analyze/src/assists/source/sort_jsx_props.rs

```diff
@@ -0,0 +1,178 @@
+    ///   active
+    ///   validate
+    ///   classes={{
+    ///     greetings: classes.greetings
+    ///   }}
+    /// />;
+    /// ```
+    ///
+    pub SortJsxProps {
+        version: "next",
+        name: "sortJsxProps",
+        language: "js",
+        recommended: false,
+        sources: &[RuleSource::EslintReact("jsx-sort-props")],
+        source_kind: RuleSourceKind::SameLogic,
+    }
+}
+
+impl Rule for SortJsxProps {
+    type Query = Ast<JsxAttributeList>;
+    type State = ();
+    type Signals = Option<Self::State>;
+    type Options = ();
+
+    fn run(_ctx: &RuleContext<Self>) -> Self::Signals {
+        Some(())
+    }
+
+    fn action(ctx: &RuleContext<Self>, _state: &Self::State) -> Option<JsRuleAction> {
+        let props = ctx.query().clone();
+        let mut non_spread_props: Option<Vec<_>> = None;
+        let mut new_props = Vec::new();
+        for prop in props.clone() {
+            match prop {
+                AnyJsxAttribute::JsxAttribute(attr) => {
+                    if let Some(non_spread_props) = &mut non_spread_props {
+                        non_spread_props.push(attr);
+                    } else {
+                        non_spread_props = Some(vec![attr]);
+                    }
+                }
+                AnyJsxAttribute::JsxSpreadAttribute(_) => {
+                    if let Some(mut non_spread_props) = non_spread_props.take() {
+                        non_spread_props.sort_by(compare_props());
+                        new_props.extend(
+                            non_spread_props
+                                .into_iter()
+                                .map(AnyJsxAttribute::JsxAttribute),
+                        );
+                    }
+                    non_spread_props = None;
+                    new_props.push(prop);
+                }
+            }
+        }
+        if let Some(mut non_spread_props) = non_spread_props {
+            non_spread_props.sort_by(compare_props());
+            new_props.extend(
+                non_spread_props
+                    .into_iter()
+                    .map(AnyJsxAttribute::JsxAttribute),
+            );
+        }
+        if new_props == props.clone().into_iter().collect::<Vec<_>>() {
+            return None;
+        }
+        let mut mutation = ctx.root().begin();
+        mutation.replace_node_discard_trivia(props, jsx_attribute_list(new_props));
+
+        Some(RuleAction::new(
+            rule_action_category!(),
+            Applicability::Always,
+            markup! { "Sort the JSX props." },
+            mutation,
+        ))
+    }
+}
+
+fn compare_props() -> impl FnMut(&JsxAttribute, &JsxAttribute) -> Ordering {
+    |a: &JsxAttribute, b: &JsxAttribute| -> Ordering {
+        let (Ok(a_name), Ok(b_name)) = (a.name(), b.name()) else {
+            return Ordering::Equal;
+        };
+        let (a_name, b_name) = (a_name.text(), b_name.text());
+
+        a_name.cmp(&b_name)
+    }
+}
```

Comment:
You should take advantage of the Rust system. Rust has `Ord` and `PartialOrd` directives. Which means that the sorting will come for free once you implement the function.

Check how we do the sorting in this rule: https://github.com/biomejs/biome/blob/main/crates/biome_json_analyze/src/assists/source/use_sorted_keys.rs

We implement the ordering here:

https://github.com/biomejs/biome/blob/d079e53e15cfd0ce95f018713b8f38d7a5341725/crates/biome_json_analyze/src/assists/source/use_sorted_keys.rs#L29-L37

Then, we use a `is_sorted` function to check if we need to run to emit an action or not: 

https://github.com/biomejs/biome/blob/d079e53e15cfd0ce95f018713b8f38d7a5341725/crates/biome_json_analyze/src/assists/source/use_sorted_keys.rs#L105-L109

Generally, this approach is better than the one you implemented because in this case, you always run the `action`, while in this case, we always run the action.

Reply:
The reason I used a separate function for comparison was that, originally, I wanted to customize it with user options, and Ord doesn't support that (?). Even though this PR implements a trimmed down version without the options, I think it still makes sense to keep this in case we want to reintroduce the option later.

However, if we aren't going to support options for assist rules anytime soon, I would follow your guidance and change it to use `Ord`

Reply:
I don't see why this wouldn't work with options. You just create a new type that holds a reference to the options other than the nodes to sort.

When implementing `Ord` and `PartialOrd`, you will read the options and change the logic accordingly